app/naive_bayes.py

```diff
@@ -0,0 +1,286 @@
+        visibility setting in advanced section of operation creation. Calculated as = (99.0% - operation visibility%). Default is 48%
+
+    Algorithm of the planner:
+        -While there are available links in the operation for live agents:
+            -If there are links that satisfy minimum operational data settings and minimum link probability of success settings:
+                -Execute them in order of likelihood of success (high to low)
+            -Run all remaining links with insufficient past operational data in atomic ordering (used by atomic planner)
+            -Drop any links that with sufficient past data but insufficient calculated probability of success
+        -Terminate when out of links or all links have too low likelihood of success.
+    """
+
+    def __init__(
+        self, 
+        operation: Operation, 
+        planning_svc: PlanningService, 
+        min_link_data : int, 
+        decision_logging : bool,
+        stopping_conditions: List[Fact] = ()
+    ):
+        """
+        :param operation:
+        :param planning_svc:
+        :param min_link_data: Minimum number runs of link required to use past statistics on it
+        :param stopping_conditions:       
+        """
+        self.operation = operation
+        self.planning_svc = planning_svc
+        self.stopping_conditions = stopping_conditions
+        self.stopping_condition_met = False
+        self.state_machine = ['bayes_state']
+        self.next_bucket = 'bayes_state'
+        self.matrix_past_links = None
+        self.min_link_data = min_link_data
+        self.decision_logging = decision_logging
+        self.min_prob_link_success = (99.0-self.operation.visibility)/(100)
+        self.links_executed = 0
+        if self.decision_logging:
+            self.planning_svc.log.debug("Operation started")   
+
+    async def execute(self):
+        await self.planning_svc.execute_planner(self)
+    
+    async def bayes_state(self):
+        """
+        Update past operational data matrix if needed then run top priority link for each agent until operation concluded
+        """
+        if self.links_executed % 10 == 0:
+            past_operation_data = await self.planning_svc.get_service('data_svc').locate('operations')
+            self.matrix_past_links = await self._build_past_links_matrix(past_operation_data)
+
+        links_to_use = []
+        for agent in self.operation.agents:
+            possible_agent_links = await self.planning_svc.get_links(operation=self.operation, agent=agent)
+            next_link = await self._get_best_link(possible_agent_links)
+            if next_link:
+                links_to_use.append(await self.operation.apply(next_link))
+        if links_to_use:
+            # Each agent will run the next available step.
+            await self.operation.wait_for_links_completion(links_to_use)
+            self.links_executed += 1
```

Comment:
Is it correct that it should only increment `links_executed` by one even if five different agents all execute a link on this iteration? If that is the correct behavior, it might be worth considering renaming it to something like `iteration_count` or similar.